src/hooks/useKeyDown.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from "react";
+
```

Comment:
For generic hooks which land in `src/hooks` please provide JS-doc style comment with use cases. If you need examples you can look up `usePreventHistoryBack` or `useSortRedirects`.

Reply:
I have a suggestion here: because atm you only need to handle an esc key just for useFullScreenMode. Following KISS rule, create only the things you need. so within useFullScreenMode just lift up that hook, call it usePressEscKey which will take the callback, and use that within useFullScreenMode. Simpler. We don't have to always build super-generic hooks because you don't know if we need such atm